src/t_hash.c

```diff
@@ -230,6 +230,12 @@ int hashTypeSet(robj *o, sds field, sds value, int flags) {
         }
         o->ptr = zl;
 
+        /*It's here to serve HINCRBY* and that in case of HSET/HMSET it'll actually be a NOP 
+         *since we have the early check before appending the first item.*/
+        if (sdslen(field) > server.hash_max_listpack_value || sdslen(value)> server.hash_max_listpack_value) {
+            hashTypeConvert(o, OBJ_ENCODING_HT);
+        }
```

Comment:
it would be more efficient to do that check before adding the item.

Reply:
also, i now realize that there's a risk (which didn't exist in your previous version), that we'll convert the encoding, but not replicate the command.
for instance, doing HSET on an existing field, after modifying the listpack threshold, we should probably make sure to handle that properly, which could either be by propagating a useless command to the replica, or moving this check back to where it was before.

Reply:
> also, i now realize that there's a risk (which didn't exist in your previous version), that we'll convert the encoding, but not replicate the command. for instance, doing HSET on an existing field, after modifying the listpack threshold, we should probably make sure to handle that properly, which could either be by propagating a useless command to the replica, or moving this check back to where it was before.

I don't quite understand. Do you mean that we will have this risk if we add the length check into the hashTypeSet?

Reply:
let's review the code and fix these bugs.
maybe the easiest thing to do is make sure `hashTypeSet` won't call `hashTypeConvert` in case nothing was really modified in the hash?